src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { getUser } from '../../api';
+import { clearCurrentTodo } from '../../features/currentTodo';
+import { useTypedSelector } from '../../app/hooks';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useTypedSelector(state => state.currentTodo);
+  const [loader, setLoader] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setLoader(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
```

Comment:
Verify that the `getUser` function is correctly implemented and imported from the `api` module. This function is essential for fetching the user details associated with the current todo.